[mpg123] Triplet VCPKG_BUILD_TYPE release fix

heydojo opened the pull request:
Fix builds which do not require
debug files.

Attempting to build mpg123 with
VCPKG_BUILD_TYPE release set in
a triplet will fail. This patch
prevent builds from failing
under VCPKG_BUILD_TYPE release
build conditions.

heydojo committed:
[mpg123] Triplet VCPKG_BUILD_TYPE release fix

Fix builds which do not require
debug files.

Attempting to build mpg123 with
VCPKG_BUILD_TYPE release set in
a triplet will fail. This patch
prevent builds from failing
under VCPKG_BUILD_TYPE release
build conditions.

PhoebeHui commented:
/azp run

ras0219-msft commented on ports/mpg123/portfile.cmake:
```diff
@@ -48,36 +48,44 @@ if(VCPKG_TARGET_IS_WINDOWS)
         PROJECT_PATH ${SOURCE_PATH}/ports/MSVC++/2015/win32/libmpg123/libmpg123.vcxproj
         RELEASE_CONFIGURATION Release${MPG123_CONFIGURATION}${MPG123_CONFIGURATION_SUFFIX}
-        DEBUG_CONFIGURATION Debug${MPG123_CONFIGURATION}${MPG123_CONFIGURATION_SUFFIX}
+        if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "debug")
```
Unfortunately, CMake's language does not work this way -- you can't put an `if()` directive inside another directive like `vcpkg_build_msbuild()`.

Instead, the `if()/endif()` will need to surround the entire `vcpkg_build_msbuild()` call or will need to set a variable which is expanded to provide arguments.

heydojo replied:
I see. Lovely. I'm pretty sure that's a quick fix I can make.
Thanks for the review!

ras0219-msft commented on ports/mpg123/portfile.cmake:
```diff
@@ -48,36 +48,44 @@ if(VCPKG_TARGET_IS_WINDOWS)
     vcpkg_build_msbuild(
```
If possible, I recommend trying to replace this with `vcpkg_install_msbuild()`[1].

[1] https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_install_msbuild.md

heydojo replied:
Please feel free to act on your advice. I have no further interest in this port past just making it work and as such pushed this to just make it work under expected conditions. Previously, the port would fail. The upshot of this being that I just removed the requirement to use mp3 playback in the application I was building. Therefore no longer requiring the output of this port.

Thanks for the recommendation though and I'm sure that some else is up to the challenge!

heydojo committed:
Remove directive inside

Unfortunately, CMake's language doesn't
support conditionals nested inside
functions according to https://github.com/microsoft/vcpkg/pull/9056#discussion_r349697844
This patch aims to resolve an attempt
to do so from the previous commit.

JackBoosY commented on ports/mpg123/portfile.cmake:
```diff
@@ -92,24 +106,26 @@ elseif(VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Darwin" OR VCPKG_CMAKE_SYSTEM_NAME STRE
+        # Debug build
+        ################
+        message(STATUS "Configuring ${TARGET_TRIPLET}-dbg")
+        vcpkg_execute_required_process(
```
We already have a simple solution here: use `vcpkg_configure_make`[1] and `vcpkg_install_make`[2].

[1] https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_configure_make.md
[2] https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/vcpkg_install_make.md

heydojo replied:
Proposed merge deleted and pull request closed.

Please update this port however you see fit. Or don't as the case may likely be.
